results/user-testing-results.md

```diff
@@ -0,0 +1,138 @@
+
+#### Flexible command palette search/filter behavior
+
+Because several participants relied heavily on the command palette to complete tasks, the limit of its filtering arose several times. From the user end, the command palette feels like it searches only for direct matches within the command list for whatever the user types in. If a user knows exactly what they are looking for, they often have success. Participants were sometimes able to quickly and flawlessly locate a desired command. But when they were exploring JupyterLab’s command options, unable to remember the exact name of a command, made a typo, or were unfamiliar with a task many participants had to give up using the command palette in order to complete the task. Depending on the assistive tech they use, typos can also be nontrivial to detect for blind or visually impaired users. Having a fuzzy search or other typo/error support rather than only the direct filter would help the command palette continue to be the boon many participants wanted it to be.
+
+## For future tests
+
+Reflecting on the findings and gaps in this set of tests, the following are questions to revisit in the future.
+
+- How might a screen reader user complete common notebook tasks in JupyterLab?
+- What are JupyterLab’s strengths and weaknesses when it comes to authoring notebooks and other documents with assistive tech?
+- What more can we learn about navigating JupyterLab, especially with a variety of assistive tech? What are good examples of navigation that support disabled users?
+- What are the full range of notebook cell output types and what feedback to disabled users have on each type?
+- What is the real time collaboration experience in JupyterLab like for assistive tech users? This question requires JupyterLab collaboration modes to be more stable before it can be tested.
+
+As mentioned in the Limits section, the sample we were able to test with for these sessions was not as comprehensive of varied disabilities and assistive tech as we would like. Recruitment for these sessions was done within existing JupyterLab community spaces and was mostly self-selected; future tests would likely benefit from a more selective sample and outreach to disability communities where appropriate. This should be done with careful consideration as the history of people who make products interrupting disability communities and asking for labor is a fraught one.
```

Comment:
```suggestion
As mentioned in the Limits section, the sample we were able to test with for these sessions was not as comprehensive of varied disabilities and assistive tech as we would like. Recruitment for these sessions was done within existing JupyterLab community spaces and was mostly self-selected; future tests would likely benefit from a more selective sample and outreach to disability communities where appropriate. This should be done with careful consideration as the history of people who make products interrupting disability communities and asking for labor is a fraught one.
```

Should a bit more be said here? or resource linked to? is it just a matter of fair compensation? or something else?

Reply:
To be honest, I did not say too much because I think this is a complicated topic I do not feel personally equipped to articulate properly. I would like to link to something, but I'm not personally aware of a link that discusses it. It also seemed unfair to potential readers to ignore it. I can easily imagine a genuine and enthusiastic person reaching out with the best intentions and still causing harm, so I felt like I had to say something.

Compensation (or a frequent request for unpaid labor) is definitely a factor. I do not think it's the entirety of the tension. 

What kind of information were you expecting in this section? I can see what I can do to provide it.

Reply:
Yeah, I wasn't sure myself whether more should be said here. I think it's okay as is. Controversy (fraught history) tends to be interesting, so I think this paragraph will pique readers' curiosity. I suspect that a lot of readers would want to know more. But it's not the point of this document, so if they want to know more, they can use a search engine, or talk to people

Reply:
I think we can leave this as is without needing more detail.
Ethical and fair user testing mainly when focusing on marginalised groups can be challenging due to many reasons beyond compensation.